float32 startX
float32 startY
float32 startZ
float32 endX
float32 endY
float32 endZ
float32 ppmm


